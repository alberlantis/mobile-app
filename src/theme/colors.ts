export default {
  BLACK: "#000",
  BLACK_INPUT: "#28292A",
  WHITE: "#FFF",
  WHITE_BOLD: "rgba(255, 255, 255, 0.66)",
  WHITE_LIGHT: "rgba(255, 255, 255, 0.33)",
  GRAY: "rgba(255, 255, 255, 0.12)",
  ORANGE_PRIMARY_DARK: "#FF7849",
  ORANGE_PRIMARY_LIGHT: "#FF8D3B",
  BLUE_PRIMARY: "#5466D8",
  TRANSPARENT: "transparent",
};
