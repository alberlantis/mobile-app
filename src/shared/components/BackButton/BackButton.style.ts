import { StyleSheet } from "react-native";
import { normalizeSize } from "src/theme";

export default StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
    width: normalizeSize(38),
    height: normalizeSize(38),
    borderRadius: normalizeSize(38) / 2,
  },
});
