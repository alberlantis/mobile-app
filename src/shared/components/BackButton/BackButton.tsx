import React from "react";
import { ColorValue, Pressable } from "react-native";
import { useNavigation, type NavigationProp } from "@react-navigation/native";

import type { UnsignedGroupParamList } from "src/navigation/UnsignedStack";
import type { SignedParamList } from "src/navigation/SignedStack";
import type { HomeTabsParamList } from "src/navigation/HomeTabs";
import Icon from "../Icon";
import { colors, fonts } from "src/theme";
import s from "./BackButton.style";

type Params = UnsignedGroupParamList & SignedParamList & HomeTabsParamList;

interface IBackButtonProps {
  color?: ColorValue;
  onPress?(): void;
}

const BackButton: React.FC<IBackButtonProps> = ({
  onPress,
  color = colors.BLACK,
}) => {
  const navigation = useNavigation<NavigationProp<Params>>();

  const handleOnPress = () => {
    if (!!onPress) onPress();
    else navigation.goBack();
  };

  return (
    <Pressable
      onPress={handleOnPress}
      style={{ ...s.container, backgroundColor: color }}
    >
      <Icon
        size={fonts[16]}
        type="Entypo"
        name="chevron-thin-left"
        color={colors.WHITE}
      />
    </Pressable>
  );
};

export default BackButton;
