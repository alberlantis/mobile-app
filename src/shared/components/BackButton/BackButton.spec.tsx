import { fireEvent, render, screen } from "@testing-library/react-native";

import type { IconProps } from "../Icon";
import colors from "src/theme/colors";
import BackButton from "./BackButton";

jest.mock("../Icon", () => {
  const { Text } =
    jest.requireActual<typeof import("react-native")>("react-native");
  return function MockIcon(props: IconProps<"Entypo">) {
    return <Text {...props}>{props.name}</Text>;
  };
});

const mockNavigation = {
  goBack: jest.fn(),
};
jest.mock("@react-navigation/native", () => ({
  useNavigation: jest.fn(() => mockNavigation),
}));

describe("BackButton", () => {
  beforeEach(() => {
    render(<BackButton />);
  });

  it("should render back button", () => {
    const button = screen.getByText("chevron-thin-left");
    expect(screen.toJSON()).toMatchSnapshot();
    expect(button).toBeOnTheScreen();
    expect(button).toHaveProp("color", colors.WHITE);
    expect(button).toHaveProp("type", "Entypo");
  });

  it("should navigate go back when tapped", () => {
    const button = screen.getByText("chevron-thin-left");
    fireEvent.press(button);
    expect(mockNavigation.goBack).toHaveBeenCalledTimes(1);
  });
});
