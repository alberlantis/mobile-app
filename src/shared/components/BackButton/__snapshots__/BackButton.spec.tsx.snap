// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton should render back button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "borderRadius": 100,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    color="#FFF"
    name="chevron-thin-left"
    type="Entypo"
  >
    chevron-thin-left
  </Text>
</View>
`;
